types/datum.go

```diff
@@ -198,6 +198,7 @@ func (d *Datum) GetBytes() []byte {
 func (d *Datum) SetBytes(b []byte) {
 	d.k = KindBytes
 	d.b = b
+	d.collation = charset.CollationBin
```

Comment:
why need set collation for bytes?

Reply:
No, set it here so that we can use `compareString` to compare them. Though I change `compareString` to `bytes.Compare` here, we can't 100% sure that we always use it rightly.

Reply:
all right